### DISCUSSION TITLE
A list of features to consider implementing

### DISCUSSION THREAD

**Discussion by DineshSolanki** [DineshSolanki/FoliCon#142] [Ideas]
### This is a list of features that can be implemented.
 please comment on this issue for new feature requests. **Do not use it to tell about bugs or errors, for that please open a new issue.**
- [ ] ~Background watcher to automatically create poster Mode icons whenever a new media in the watched folder is added.~
- [x] Save id of search in a file so that in case of a system failure or shutdown, the user won't have to go through selecting ambiguous titles, as id would find the exact title.
- [x] Globalization of FoliCon
- [x] Ability to browse through and select a different poster in the poster mode.
- [x] option to give a custom rating on the rating badge.
- [x] display the corresponding image, just before the text http://prntscr.com/t1l5rc
- [x] Support for trailer or videos on the Search screen.
- [ ] Support for TV-season icons and auto-detection.
- [ ] More rating badges.
- [ ] ~~Show media quality on icon(if available)~~
- [ ]  Feature to rename folders to correct media files. 
- [x] Dark mode

**Comment by FazCodeFR** [comment]
Hello ! Nice soft ! 
- [x] display the corresponding image, just before the text http://prntscr.com/t1l5rc

- [x] works with video series

**Comment by DineshSolanki** [comment]
hey, thanks for suggesting, 
for the first feature, I initially worked on this, as I thought it would help in quick selection, but as the image will be downloaded for each result, it will be an unnecessary overhead, as an alternative, I can load images only of those which item is clicked but even then I'll have to send a request, so I don't find this feasible, correct me if there's something.

and for the second suggestion, I couldn't understand what you mean by video series, can you please explain more about it.

**Comment by DineshSolanki** [comment]
I tested the first feature today and I think it doesn't need to be an overhead as I only gave poster path to the selected item as a string, and image Control being smart enough automatically downloads it when needed so no need to request again

**Comment by DineshSolanki** [comment]
Poster preview in Poster mode result introduced in 0d84b5451d1fbd04765ed9d82363c7249283ebcb
And available in the latest release

**Comment by DineshSolanki** [comment]
About background watcher, I think it's not needed really, considering most of us don't want to let the app run in the background the whole time.

**Comment by pierx** [comment]
Regarding the new feature the custom icon setter, I think it would be nice if it target the exact single folder we would like to change and drag the exact icon directly into the box. 
Also it would be good if the desktop.ini IconResource= doesn't include the path but directly the icon name so the folder can be move freely, just like the normal icon mode.
Here's mockup for it
![folicon_request](https://user-images.githubusercontent.com/11854637/105280168-1cdf4f00-5be4-11eb-87cb-149cb2ec79c8.png)

**Comment by DineshSolanki** [comment]
@pierx sorry for the late reply, I was a little occupied.
Thank you for pointing that out, I have corrected that now, 
also about the "one folder". you can delete other folders and icons keeping only the ones that you want, use the delete key. or other usual shortcuts such as "Shift + End" for selecting to end. 
you can also rearrange the items.
![FoliCon single folder](https://user-images.githubusercontent.com/15937452/105525862-a3408180-5d07-11eb-8ea5-fb2b82a8b1ae.gif)

**Comment by DineshSolanki** [comment]
@pierx v3.2.1 available with "icon path fix"

**Comment by pierx** [comment]
> @pierx sorry for the late reply, I was a little occupied.
> Thank you for pointing that out, I have corrected that now,
> also about the "one folder". you can delete other folders and icons keeping only the ones that you want, use the delete key. or other usual shortcuts such as "Shift + End" for selecting to end.
> you can also rearrange the items.
> ![FoliCon single folder](https://user-images.githubusercontent.com/15937452/105525862-a3408180-5d07-11eb-8ea5-fb2b82a8b1ae.gif)

Thanks for the tips and the update!

**Comment by DineshSolanki** [comment]
feature -Ability to browse through and select a different poster in the poster mode.
added in version 3.3, looking forward for feedback.

https://user-images.githubusercontent.com/15937452/126371572-c4395a2c-8c22-49e7-ae91-9143550cb1da.mp4

**Comment by pierx** [comment]
Thanks for the update!
Suggestion, instead of clicking the poster for option, it might better to add button for that action.
![foli](https://user-images.githubusercontent.com/11854637/126479619-8ca30dc9-f3c9-4268-a051-aea0ffffd671.jpg)
By the way, this poster option doesn't seem to work on TV search mode.

**Comment by DineshSolanki** [comment]
> 
> 
> Thanks for the update!
> Suggestion, instead of clicking the poster for option, it might better to add button for that action.
> ![foli](https://user-images.githubusercontent.com/11854637/126479619-8ca30dc9-f3c9-4268-a051-aea0ffffd671.jpg)
> By the way, this poster option doesn't seem to work on TV search mode.

I Tried it on TV mode, It's working, but I'll take a look again at the possible causes, also this is yet not available for game mode.
about the button, Yes I should do, maybe next weekend, Thanks,

**Comment by DineshSolanki** [comment]
@pierx TV mode issue is fixed, and show more button is also added in the latest release. please download the same.
Also I want to take a moment to thank you for your valuable comments and suggestion, they are a big reason for FoliCon's evolution.

**Comment by pierx** [comment]
> @pierx TV mode issue is fixed, and show more button is also added in the latest release. please download the same.
> Also I want to take a moment to thank you for your valuable comments and suggestion, they are a big reason for FoliCon's evolution.

Thank you very much for the update. The show more button works nicely and the TV mode issue is fixed now.

**Comment by DineshSolanki** [comment]
Custom rating now also available, and more tool-tips.

**Comment by DineshSolanki** [comment]
4 new languages added, although in pre-release, needs more testing.
Download from release section.

**Comment by DineshSolanki** [comment]
ID of picked titles now stored, for resolving titles automatically.

**Comment by riyaddl** [comment]
Hello, Great app. 

- [ ] the ability to add a custom overlay.
- [ ] the ability to change the rating badge (with a star for example)

**Comment by DineshSolanki** [comment]
> Hello, Great app. 
> 
> - [ ] the ability to add a custom overlay.
> - [ ] the ability to change the rating badge (with a star for example)

Will look into them, in the meantime you might put here the custom overlays you fancy, so That I can aseess if they can be made into folicon

**Comment by tiwey15** [comment]
Love the app! I've been trying to find something like this for a while. Thank you for your hard work.

I do have two suggestions.

1) the DVD Icon is great, but itd be really cool if the icon could change based on the resolution of the file. I.e. if it's 4k replace "DVD Video" with "UHD" or 1080p with "HD" and everything else "SD".

2) Also, the name search funtion is really good, i wonder if it'd be too much trouble to have it rename the folder along with changing the icon.

Thanks so much. You are wonderful.

**Comment by DineshSolanki** [comment]
Hey, thanks for the kind words and support through Paypal, truly appreciated it,
About the 'DVD' on layout, that is not generated by 'FoliCon' it's an icon that I used from DeviantArt, FoliCon would have trouble generating mockups like photoshop but I tried to get closer to that. **and you can choose from other layouts as well, there are 5 of them in-app**
![image](https://user-images.githubusercontent.com/15937452/147857849-8c59d4a9-ac83-4f22-8dec-d68b2c5f1ffe.png)

But we might be able to achieve this if I manually modify icons with photoshop and try to identify quality from the name of the media file.

About the second one, renaming shouldn't be a problem, but there are tonnes of apps that do it better already, **I would recommend Filebot 4.7**

I'll add them as future enhancements -
1. Show media quality on icon
2. Feature to rename folders to correct media files.

**Comment by abewartech** [comment]
integrated with : 
- TinyMediaManager
- Radarr & Sonarr

**Comment by DineshSolanki** [comment]
> integrated with :
> 
>     * TinyMediaManager
> 
>     * Radarr & Sonarr

How should I go with that, AFAIK they don't have a plugin system, do you mean FoliCon should carry feature from them?

**Comment by TheFmC** [comment]
Thank You For Your Work i really like the app .. but their is one problem iam facing .. low resolution for icons

**Comment by DineshSolanki** [comment]
> Thank You For Your Work i really like the app .. but their is one problem iam facing .. low resolution for icons

Hi @TheFmC Issue has been solved, will be releasing soon, thanks for reporting.

**Comment by TheFmC** [comment]
> > Thank You For Your Work i really like the app .. but their is one problem iam facing .. low resolution for icons
> 
> Hi @TheFmC Issue has been solved, will be releasing soon, thanks for reporting.

Thank you @DineshSolanki for solving the problem in a short time
I can asure that it work now <3

**Comment by MasoudRahmani** [comment]
Anime? 🥹

**Comment by DineshSolanki** [comment]
> Anime? 🥹

Doesn't anime works with professional mode?

**Comment by MasoudRahmani** [comment]
> > Anime? 🥹
> 
> Doesn't anime works with professional mode?

Yes it does actually, also with data from IMDb, but i thought since it was not mentioned maybe there were more to it.
like an API [Anilist](https://docs.consumet.org/rest-api/Meta/anilist-anime/get-anime-info)

[Comic Books](https://metron.cloud/) ,

[Manga](https://docs.consumet.org/rest-api/Meta/anilist-manga/info)

**Reply by DineshSolanki** [reply]
@MasoudRahmani if FoliCon were to search comicbooks, would it need to search for issue or search for volumes? which one

**Reply by MasoudRahmani** [reply]
> @MasoudRahmani if FoliCon were to search comicbooks, would it need to search for issue or search for volumes? which one

I personally use issue.

**Comment by DineshSolanki** [comment]
> > > Anime? 🥹
> > 
> > Doesn't anime works with professional mode?
> 
> Yes it does actually, also with data from IMDb, but i thought since it was not mentioned maybe there were more to it.
> like an API [Anilist](https://docs.consumet.org/rest-api/Meta/anilist-anime/get-anime-info)
> 
> [Comic Books](https://metron.cloud/) ,
> 
> [Manga](https://docs.consumet.org/rest-api/Meta/anilist-manga/info) 
 
The APIs you provided seems promising. 
Will surely look into this information.

**Comment by DineshSolanki** [comment]
Pre-release available with dark mode, check it out, and help me get it release ready by your suggestions and feedback everyone https://github.com/DineshSolanki/FoliCon/releases/tag/v3.11.0-beta.1
> update - available in v3.11.0

**Reply by MasoudRahmani** [reply]
thanks

**Comment by DineshSolanki** [comment]
Dropping the following feature from roadmap -

-  'Background watcher to automatically create poster Mode icons whenever a new media in the watched folder is added.' 
- Show media quality on icon
> The video files inside the folder can be changed  anytime, but the folder icon will remain, until generated again, causing the icon to be closely tied to its content. 


Background watcher would need the application to be running all the time. this is odd for a desktop application and the frequency of new media being added in the folder.

**Comment by DineshSolanki** [comment]
@pierx @MasoudRahmani @abewartech @tiwey15 - I'm thinking the next feature should be the support for TV season search.

I have created a basic theory of how it would be, and I'm looking forward for your suggestions, ideas, designs, any kind of insight - 

- If user has selected TV mode, in poster search, and the selected folder contains sub-directories with the following name pattern - 'Season X'
- then FoliCon will show the user a confirmation that the selected folder is recognized as a TV series folder which contains season of the same. and the user have to chose if user want to process the folder as TV season folder or normal way.
- If the user confirms then FoliCon searches for the TV series title, if the title is ambiguous or has multiple results, then user will be shown a drop-down where user can select the correct title, once selected, user will be taken to a new window, similar to the current poster mode search window.
but instead of searching one by one, it will show all detected season on the same window in single list, each result will be associated with its corresponding directory, 
- User can see poster and chose from more available folder as in the normal way, the trailer/videos will be shown.
- But instead of choosing for each folder one by one, all will be in same list, on clicking continue, the selected poster and rating will be associated will those folders and the poster will be created

**Comment by MasoudRahmani** [comment]
**Thank you**, this is a great idea but since it might take some time to process, i'd like an option to check "**find seasonal poster**" or **not**.
This way you don't have to show confirmation for each series and simplify?

for changing mid way (changed my mind) or recommendation for auto mode maybe a flashing warning like this.
![image](https://github.com/DineshSolanki/FoliCon/assets/5321268/46b62cd7-8cab-48a1-ba26-f086cf08e285)

I have no idea what limitation or requirement are This was the way i found easier? maybe.  sorry ❤️

These are some patterns that i found when sorting season and they i think "season x" might not be enough.

Also some might only download one season and the season name would be in root folder.

#### (fullname)
```
"\Seinfeld S Season 01 720p QAAC HDTV x265 hevc P11"
"\Seinfeld S Season 02 720p QAAC HDTV x265 hevc P11"
```
#### (Long running show seperated by year)
```
"2018"
"2019"
"(2020)"
```
#### (Numbering)
```
"01"
"02"
"\...Extras"
"\S09"
"\S10"
"\S010"
```

**Reply by DineshSolanki** [reply]
@MasoudRahmani certainly these are useful thoughts, specially the naming pattern, we can try to support as much as possible, ofcourse it needs a good planning on development front.

Will. Keep you all updated

**Comment by DineshSolanki** [comment]
FoliCon docs are now live - https://dineshsolanki.github.io/FoliCon-docs/

**Comment by ArunaTelshan** [comment]
Greatly appreciate if you could add folder renaming feature.

**Reply by DineshSolanki** [reply]
> Greatly appreciate if you could add folder renaming feature.

Hi, I would suggest filebot for that, filebot 4.7.8 is free

**Comment by ayan312002** [comment]
Support for icon creation of a folder? 
I might be missing something but Folicon doesn't seem to support creating icon for a specific folder. Instead of creating icons for directories inside a parent folder, it would be nice if you could select that folder itself to create icons.

**Comment by DineshSolanki** [comment]
Thinking about some new UI 🤔
![Screenshot 2025-01-10 000103](https://github.com/user-attachments/assets/1aa6fad8-1fa8-433a-aeb7-05f58643586c)

![image](https://github.com/user-attachments/assets/857eb78f-5e9d-4fe5-a80c-2b79c7a4ce70)

<img width="1024" height="559" alt="image" src="https://github.com/user-attachments/assets/c9fbe907-2699-4537-8174-a0dff532bf81" />
